exchanges/bybit/bybit_usdcfutures.go

```
@@ -0,0 +1,1143 @@
+}
+
+// GetUSDCPredictedFundingRate gets predicted funding rates and my predicted funding fee.
+func (by *Bybit) GetUSDCPredictedFundingRate(ctx context.Context, symbol currency.Pair) (predictedFundingRate, predictedFundingFee float64, err error) {
+	resp := struct {
+		Result struct {
+			PredictedFundingRate float64 `json:"predictedFundingRate,string"`
+			PredictedFundingFee  float64 `json:"predictedFundingFee,string"`
+		} `json:"result"`
+		USDCError
+	}{}
+
+	req := make(map[string]interface{})
+	var symbolValue string
+	if !symbol.IsEmpty() {
+		symbolValue, err = by.FormatSymbol(symbol, asset.USDCMarginedFutures)
+		if err != nil {
+			return resp.Result.PredictedFundingRate, resp.Result.PredictedFundingFee, err
+		}
+		req["symbol"] = symbolValue
+	} else {
+		return resp.Result.PredictedFundingRate, resp.Result.PredictedFundingFee, errSymbolMissing
+	}
+
+	err = by.SendUSDCAuthHTTPRequest(ctx, exchange.RestUSDCMargined, http.MethodPost, usdcfuturesGetPredictedFundingRate, req, &resp, usdcGetPredictedFundingRate)
+	predictedFundingRate = resp.Result.PredictedFundingRate
+	predictedFundingFee = resp.Result.PredictedFundingFee
+	return
+}
+
+// SendUSDCAuthHTTPRequest sends an authenticated HTTP request
+func (by *Bybit) SendUSDCAuthHTTPRequest(ctx context.Context, ePath exchange.URL, method, path string, data interface{}, result UnmarshalTo, f request.EndpointLimit) error {
+	creds, err := by.GetCredentials(ctx)
+	if err != nil {
+		return err
+	}
+
+	if result == nil {
+		result = &USDCError{}
+	}
+
+	endpointPath, err := by.API.Endpoints.GetURL(ePath)
+	if err != nil {
+		return err
+	}
+
+	err = by.SendPayload(ctx, f, func() (*request.Item, error) {
+		nowTimeInMilli := strconv.FormatInt(time.Now().UnixMilli(), 10)
+		headers := make(map[string]string)
+		var payload, hmacSigned []byte
+
+		if data != nil {
+			if d, ok := data.(map[string]interface{}); ok {
```

if data assertion fails, we should return an error to the caller.